File: samples/java/SupportChat/Server/Support/src/main/java/systems/zlink/samples/supportchat/server/support/spots/conversationspot/ConversationIdleTimerHandler.java
package systems.zlink.samples.supportchat.server.support.spots.conversationspot;

import systems.zlink.framework.spots.ZLinkSpotTimerHandler;
import systems.zlink.framework.spots.ZLinkTimerTick;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

// --8<-- [start:doc-sc-idle-timer]
public final class ConversationIdleTimerHandler implements ZLinkSpotTimerHandler<ConversationSpot> {
    @Override
    public CompletionStage<Void> handle(ConversationSpot spot, ZLinkTimerTick tick) {
        spot.checkIdle();
        return CompletableFuture.completedFuture(null);
    }
}
// --8<-- [end:doc-sc-idle-timer]

File: samples/kotlin/SupportChat/Server/Support/src/main/kotlin/systems/zlink/samples/kotlin/supportchat/server/support/infrastructure/zlink/spots/conversationspot/handlers/ConversationIdleTimerHandler.kt
package systems.zlink.samples.kotlin.supportchat.server.support.infrastructure.zlink.spots.conversationspot.handlers

import systems.zlink.framework.kotlin.ZLinkSuspendingSpotTimerHandler
import systems.zlink.framework.spots.ZLinkTimerTick
import systems.zlink.samples.kotlin.supportchat.server.support.infrastructure.zlink.spots.conversationspot.ConversationSpot

class ConversationIdleTimerHandler : ZLinkSuspendingSpotTimerHandler<ConversationSpot> {
    override suspend fun handle(spot: ConversationSpot, tick: ZLinkTimerTick) {
        spot.checkIdle()
    }
}
